Switch to FB share button from Like button

ddohler opened the pull request:
## Overview

Swaps out the Facebook Like buttons on the homepage and plan Sharing page for FB Share buttons instead.

### Checklist

- [x] PR has a name that won't get you publicly shamed for vagueness
- [x] Files changed in the PR have been `yapf`-ed for style violations

### Demo

<img width="169" alt="screen shot 2018-04-26 at 12 25 05 pm" src="https://user-images.githubusercontent.com/447977/39318723-e13ea484-494c-11e8-9df2-266c050a161c.png">


### Notes

- Doing this uncovered that the domain name isn't getting properly set in `settings.py`. We currently always set `ALLOWED_HOSTS` to `['web']`, which is what results from running via docker-compose, but this means that whenever Django generates a hostname, it ends up as `http://web/whatever/` which is incorrect. We're going to need to make sure that we can tell Django what domain it is running on; I've added [#157125912](https://www.pivotaltracker.com/story/show/157125912) for this.
- The dialog that pops up when you click this button always closes immediately, whether or not you are logged into Facebook. I suspect that this is because FB is detecting the invalid domain name problem and bailing out.

## Testing Instructions

 * Navigate to the homepage and confirm that there is a FB share button available.
 * View source and confirm that the `og:*` meta tags in the header contain appropriate values
 * Share a plan and then visit its Share page, and confirm that there is a FB share button available there as well.
 * View source and confirm that the `og:*` meta tags contain appropriate values here; specifically, the title should include the legislative body, plan name, and username.

Closes #156479159

ddohler commented:
I've rebased this on develop now that the Django hostname fix is merged and I think it's ready for another look.

ddohler commented:
I've put in a fix for the FB button on the share page, it uses a similar trick to the Twitter button now.

shreshthkhilani commented on django/publicmapping/redistricting/templates/viewplan.html:
```diff
@@ -209,6 +211,15 @@
 
   </head>
   <body onload="init();" id="app">
+    <!-- Load Facebook SDK for JavaScript -->
```
Can this Facebook SDK be templated out into another file and just dropped in here instead of copied in both places?

ddohler replied:
Good call! I did this, but in the process noticed that we're not using template inheritance, which makes me 😢

shreshthkhilani commented on django/publicmapping/redistricting/templates/viewplan.html:
```diff
@@ -1219,7 +1244,7 @@ <h3>{% blocktrans %}You can select up to three statistics at once to display in
             if ($(this).tabs('option', 'selected') == 3) {
                 exportIndexFile.init();
-                createTwitterButton();
+                createShareButtons();
```
When you resolve the merge conflict you have, you'll notice that 
```js
createTwitterButton();
```
was changed to 
```js
if (typeof createTwitterButton === 'function') {
  createTwitterButton();
}
```
This is because that function is `undefined` if the plan isn't yet shared and throws a console error. Your implementation is throwing the same error and I recommend resolving merge conflicts and using the change similarly.

ddohler commented:
I rebased on `develop` and switched over the console error fix to use the new function name. I also split out the facebook snippet into its own template.
